src/OpenTracing.Noop/NoopScopeManager.cs

```diff
@@ -0,0 +1,46 @@
+namespace OpenTracing.Noop
+{
+    /// <summary>
+    /// A noop (i.e., cheap-as-possible) implementation of an <see cref="IScopeManager"/>.
+    /// </summary>
+    public sealed class NoopScopeManager : IScopeManager
```

Comment:
It could, but having a public type allows unit tests etc. to test for these specific types (e.g. `if (tracer is NoopTracer)`. I guess that's also the reason why the Java version has the public interfaces.

Reply:
I’d discourage anyone doing an if (type) as it violates LSP of SOLID :(

Per the tests, the common approach is to use InternalsVisibleTo

Reply:
I made all types internal. We could always go from internal->public later should someone really need it. The other way around is harder.